core/lib/contract_verifier/src/compilers/solc.rs

```diff
@@ -128,7 +132,11 @@ impl Compiler<SolcInput> for Solc {
         if output.status.success() {
             let output = serde_json::from_slice(&output.stdout)
                 .context("zksolc output is not valid JSON")?;
-            parse_standard_json_output(&output, input.contract_name, input.file_name, true)
+            let output =
+                parse_standard_json_output(&output, input.contract_name, input.file_name, true)?;
+            let id =
+                ContractIdentifier::from_bytecode(BytecodeMarker::Evm, output.deployed_bytecode());
```

Comment:
It seems that in all `Compiler` impls, the ID is constructed from `output.deployed_bytecode()` + bytecode type. If this is the case, can't ID construction be moved up the call chain? This would allow retaining a simpler `Compiler` interface.